pkg/controller/authentication/authentication_controller_test.go

```diff
@@ -12,24 +12,25 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package authentication
+package authentication_test
```

Comment:
Do you feel it is important to switch this to *_test?
I probably recently changed this to without the _test because I think testing of atleast one of the other controllers needed access to internals and so I was trying to make them all consistent to the _test omitted form.
Is there a downside to the non-_test form that we should avoid it?

Reply:
There is a convenience method for these internals. I guessed you did not know when you added that test and I refactored that. I'm good either way.

Reply:
I'll revert.

Reply:
So I'm not completely opposed to having the same package for the tests, but the downside is specifically that you are allowed to access the internals, and you lose the black box testing. The best unit tests usually only use interfaces that are exposed outside of the package, mimicing how the package would actually be used.